### ISSUE TITLE
I can't reproduce the accuracy of yolo11n.yaml on COCO2017.

### ISSUE THREAD

**Issue by ctx289** [ultralytics/ultralytics#20630]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I can't reproduce the accuracy of yolo11n.yaml on COCO2017 when training with 8 NVIDIA V100 GPUs. Here are my training environment and parameters. I'm wondering if there are any problems that lead to the misalignment of the accuracy. My PyTorch isn't the latest version, but I don't think it will have a significant impact on the training accuracy. I've looked through the documentation, but I can't seem to find the specific training configuration used when we train YOLO11. I hope you can help me solve this problem.Thank you very much for your answer.

<img width="310" alt="Image" src="https://github.com/user-attachments/assets/f90704c8-e240-4349-8307-aa928b05393a" />

model=YOLO('/fuxi_team3_intern/persons/sweethxchen/ultralytics/ultralytics/cfg/models/11/yolo11n.yaml')
model.train(data='/fuxi_team3_intern/persons/sweethxchen/ultralytics/ultralytics/cfg/datasets/coco.yaml',
            epochs=500, imgsz=640, batch=128, device=[0,1,2,3,4,5,6,7])

python3 -m torch.distributed.run --nproc_per_node 8 train.py --device 0,1,2,3,4,5,6,7 

epochs: 500
time: null
patience: 100
batch: 128
imgsz: 640
save: true
save_period: -1
cache: false
device:
- 0
- 1
- 2
- 3
- 4
- 5
- 6
- 7
workers: 8
project: null
name: train5
exist_ok: false
pretrained: true
optimizer: auto
verbose: true
seed: 0
deterministic: true
single_cls: false
rect: false
cos_lr: false
close_mosaic: 10
resume: false
amp: true
fraction: 1.0
profile: false
freeze: null
multi_scale: false
overlap_mask: true
mask_ratio: 4
dropout: 0.0
val: true
split: val
save_json: false
conf: null
iou: 0.7
max_det: 300
half: false
dnn: false
plots: true
source: null
vid_stride: 1
stream_buffer: false
visualize: false
augment: false
agnostic_nms: false
classes: null
retina_masks: false
embed: null
show: false
save_frames: false
save_txt: false
save_conf: false
save_crop: false
show_labels: true
show_conf: true
show_boxes: true
line_width: null
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: true
opset: null
workspace: null
nms: false
lr0: 0.01
lrf: 0.01
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 0.5
dfl: 1.5
pose: 12.0
kobj: 1.0
nbs: 64
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
bgr: 0.0
mosaic: 1.0
mixup: 0.0
cutmix: 0.0
copy_paste: 0.0
copy_paste_mode: flip
auto_augment: randaugment
erasing: 0.4
crop_fraction: 1.0
cfg: null
tracker: botsort.yaml

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ctx289, thank you for your detailed question and for sharing your training setup with Ultralytics 🚀! This is an automated response to help guide you while an Ultralytics engineer reviews your issue.

We recommend checking out the [Docs](https://docs.ultralytics.com/) for reference, especially the [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, as many common questions are addressed there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) if possible. This helps us understand and debug the issue efficiently.

For custom training or reproducibility questions, please ensure you:
- Share your dataset configuration and any relevant dataset samples
- Attach complete training logs, especially where discrepancies appear
- Confirm you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)

Join our community for more support and discussion! For real-time chat, visit [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth topics, check [Discourse](https://community.ultralytics.com/). Or join the conversation on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Please ensure you’re using the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to rule out any issues resolved in recent releases:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in the following up-to-date, verified environments (all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) are preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu every 24 hours and on every commit.

An Ultralytics engineer will review your issue and follow up soon. Thank you for helping us improve YOLO!

**Comment by ctx289** [comment]
I'm using the version of Ultralytics that is later than 8.3.100.

**Comment by ctx289** [comment]
I'm using the version of Ultralytics that is later than 8.3.100. The mAP (mean Average Precision) of 0.5:0.95 that I tested is only 38.2, which has a significant difference.

**Comment by Y-T-G** [comment]
You can read this

https://github.com/ultralytics/ultralytics/issues/17048#issuecomment-2424742030

**Comment by ctx289** [comment]
> You can read this
> 
> [#17048 (comment)](https://github.com/ultralytics/ultralytics/issues/17048#issuecomment-2424742030)

Thank you very much for your answer. Here I would like to ask about the training parameter "copy_paste: 0.1 # (float) segment copy-paste (probability)". Do I need to add any additional datasets? How will it be utilized in the training experiment on COCO2017?

**Comment by Y-T-G** [comment]
No. You just use the one that comes with Ultralytics `coco.yaml`.

**Comment by ctx289** [comment]
# Ultralytics 🚀 AGPL-3.0 License - https://ultralytics.com/license

# COCO 2017 dataset https://cocodataset.org by Microsoft
# Documentation: https://docs.ultralytics.com/datasets/detect/coco/
# Example usage: yolo train data=coco.yaml
# parent
# ├── ultralytics
# └── datasets
#     └── coco  ← downloads here (20.1 GB)

# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path:  # dataset root dir
train: /fuxi_team3_intern/data/coco/images/train2017 # train images (relative to 'path') 118287 images
val: /fuxi_team3_intern/data/coco/images/val2017 # val images (relative to 'path') 5000 images
test: /fuxi_team3_intern/data/coco/images/val2017 # 20288 of 40670 images, submit to https://competitions.codalab.org/competitions/20794
# Classes
names:
  0: person
  1: bicycle
  2: car
 ............
I downloaded the COCO dataset myself and downloaded the annotations from coco2017labels.zip at the link https://github.com/ultralytics/assets/releases/download/v0.0.0/coco2017labels.zip provided by Ultralytics. This means that the current annotations only contain category and bounding box information. Can the copy_paste method here obtain mask information? If not, how should I process the data now? Thank you for your answer.

> No. You just use the one that comes with Ultralytics `coco.yaml`.

**Comment by ctx289** [comment]
path:  # dataset root dir
train: /fuxi_team3_intern/data/coco/images/train2017 # train images (relative to 'path') 118287 images
val: /fuxi_team3_intern/data/coco/images/val2017 # val images (relative to 'path') 5000 images
test: /fuxi_team3_intern/data/coco/images/val2017 # 20288 of 40670 images, submit to https://competitions.codalab.org/competitions/20794

# Classes
names:
  0: person
  1: bicycle
  2: car
..............
I downloaded the COCO dataset myself and downloaded the annotations from coco2017labels.zip at the link https://github.com/ultralytics/assets/releases/download/v0.0.0/coco2017labels.zip provided by Ultralytics. This means that the current annotations only contain category and bounding box information. Can the copy_paste method here obtain mask information? If not, how should I process the data now? Thank you for your answer.

**Comment by ctx289** [comment]
> No. You just use the one that comes with Ultralytics `coco.yaml`.

path: # dataset root dir
train: /fuxi_team3_intern/data/coco/images/train2017 # train images (relative to 'path') 118287 images
val: /fuxi_team3_intern/data/coco/images/val2017 # val images (relative to 'path') 5000 images
test: /fuxi_team3_intern/data/coco/images/val2017 # 20288 of 40670 images, submit to https://competitions.codalab.org/competitions/20794

Classes

names:
0: person
1: bicycle
2: car
..............
I downloaded the COCO dataset myself and downloaded the annotations from coco2017labels.zip at the link https://github.com/ultralytics/assets/releases/download/v0.0.0/coco2017labels.zip provided by Ultralytics. This means that the current annotations only contain category and bounding box information. Can the copy_paste method here obtain mask information? If not, how should I process the data now? Thank you for your answer.

**Comment by ctx289** [comment]
The paths of my images and annotations are respectively coco/images/train2017 and the corresponding coco/labels/train2017. What should I modify?

**Comment by ctx289** [comment]
Oh, I see. When we actually carry out the YOLO detection training task, we use the segmentation labels of COCO2017, right?

**Comment by Y-T-G** [comment]
Yes

**Comment by ctx289** [comment]
> Yes

Thank you very much! Your reply is of great importance to me.

**Comment by ctx289** [comment]
> Yes

By the way, so should I use segmentation annotations instead of detection annotations for both train and val?

**Comment by glenn-jocher** [comment]
Yes, for training YOLO detection models on COCO to reproduce the published accuracy metrics, you should use the segmentation annotations for both train and val sets. The official COCO.yaml configuration automatically downloads the segment-based annotations via:

```python
segments = True  # segment or box labels
...
urls = [url + ("coco2017labels-segments.zip" if segments else "coco2017labels.zip")]
```

This is because the augmentations used during training (especially copy_paste at 0.1 probability) utilize the segmentation masks for more effective data augmentation, even when the final task is detection. The segmentation masks allow for more precise copying of objects between images during training.

Make sure your dataset follows the structure in the COCO.yaml file, with images in the appropriate directories and segmentation labels properly organized. This should help you reproduce the expected accuracy metrics.

**Comment by ctx289** [comment]
> 是的，为了在COCO上训练YOLO检测模型以重现已发布的准确性指标，您应该为训练集和val集使用分割注释。官方COCO.yaml配置通过以下方式自动下载基于分段的注释：
> 
> segments = True  # segment or box labels
> ...
> urls = [url + ("coco2017labels-segments.zip" if segments else "coco2017labels.zip")]
> 这是因为训练期间使用的增强（特别是0.1概率的copy_paste）利用分割掩码进行更有效的数据增强，即使最终任务是检测。分割掩码允许在训练期间更精确地复制图像之间的对象。
> 
> 确保您的数据集遵循COCO.yaml文件中的结构，适当组织适当目录中的图像和分割标签。这应该有助于您重现预期的准确性指标。

from ultralytics import YOLO, ASSETS

model = YOLO("yolo11x.pt")
print(model.ckpt["train_args"])

Okay, thank you very much for your answer. By the way, I noticed that YOLO11n-m uses a single GPU with a batch size of 128, YOLO11l uses 2 GPUs with a batch size of 128 per GPU, and YOLO11x uses 3 GPUs with a batch size of 128 per GPU. Will this training result be better? Why not uniformly use 8 V100 GPUs to speed up? Thank you for your excellent work, and your reply is very helpful to me.

**Comment by glenn-jocher** [comment]
This is an excellent observation about the GPU configurations. The different GPU setups for model variants are intentional and based on optimizing for model performance rather than just speed.

There are several reasons for this approach:

1. Model convergence is sensitive to effective batch size. While using 8 GPUs would accelerate training, it could negatively impact final model accuracy by creating an excessively large effective batch size.

2. Larger models like YOLO11x require more GPU memory per sample, so distributing across multiple GPUs allows maintaining appropriate per-GPU batch sizes.

3. The training configurations were carefully tuned for each model size to achieve optimal accuracy. Changing the GPU count without adjusting other hyperparameters might lead to worse results.

You could certainly experiment with 8 GPUs for faster training, but you'd need to carefully adjust learning rates and other hyperparameters to maintain convergence quality. The published configurations prioritize reproducibility and optimal accuracy over raw training speed.

**Comment by ctx289** [comment]
> 这是对GPU配置的一个很好的观察。模型变体的不同GPU设置是有意的，基于优化模型性能，而不仅仅是速度。
> 
> 这种方法有几个原因：
> 
> 1. 模型收敛对有效批次大小很敏感。虽然使用8个GPU会加速训练，但它可能会通过创建过大的有效批次大小而对最终模型的准确性产生负面影响。
> 2. 像YOLO11x这样的大型模型每个样本需要更多的GPU内存，因此在多个GPU上分布可以保持每个GPU的适当批次大小。
> 3. 训练配置针对每个模型尺寸进行了仔细调整，以达到最佳准确性。在不调整其他超参数的情况下更改GPU计数可能会导致更差的结果。
> 
> 你当然可以用8个GPU来尝试更快的训练，但你需要仔细调整学习率和其他超参数来保持收敛质量。发布的配置优先考虑可重复性和最佳准确性，而不是原始训练速度。

Okay, thank you for your answer. Yes, I've also noticed this issue. Although single-GPU training takes a significantly longer time, it yields better results. In contrast, multi-GPU training with the same configuration leads to premature convergence.
